### PR TITLE
Migrate to the latest base and improve `BackendClient` API

### PR DESCRIPTION
**Pull request by dmytro-grankin**
This PR is a counterpart of the [improvements](https://github.com/SpineEventEngine/base/pull/274) in `base`.

#### Backend client
The API of `BackendClient` was improved and not it doesn't require a user to pass `TypedMessage`. Instead, Protobuf `Message`s should be passed to the API directly.

Before:
```js
const command = new CreateTask();
const typedCommand = new TypedMessage(command, 'type.spine.io/spine.web.task.CreateTask');
backendClient.sendCommand(typedCommand, ...);
```

After:
```js
const command = new CreateTask();
backendClient.sendCommand(command, ...);
```

The same changes were made in the API for querying and subscriptions.

Also, now it is required to explicitly register Protobuf types to be used:

```js
import * as webTypes from './web/proto/index.js';
BackendClient.registerTypes(webTypes, anotherTypes);
```

The registration is mainly required to properly parse Protobuf messages.

#### NPM artifacts

Previously two artifacts were published to NPM, `spine-web-client-proto` and `spine-web-client`. It was decided to merge these artifacts since it is unlikely that they will be used separately.

Also, the `spine-web-client` artifact was renamed to `spine-web`.

Protobuf types now can be imported like `spine-web/proto/spine/core/user_id_pb.js`.

And the most important is that published code is now transpiled to ES5 (previous versions were in ES6). Source maps are provided as well so it will be possible to debug the library using original files written in ES6.

### Internal changes

- The structure of `client-js` module was changed. Now, generated Protobufs are placed near the sources, under the `client-js/main/proto` for production types and under the `client-js/test/proto` for test types. The main for the reason is to simplify publishing to NPM.
- `KnownTypes` and `TypeParsers` classes were introduced. They accumulate all the Protobuf types used by the backend client.
- Aliases for imports in test files were added. It allows to simplify imports and perform project structure changes easier. Aliases were not introduced for non-test files to avoid the necessity to transpile code before publishing to NPM.
- Build scripts for `client-js` were improved to increase the build speed and make project structure changes easier.

**TBD**
- [ ] Deprecate `spine-web-client` in the NPM registry.
- [ ] Deprecate `spine-web-client-proto` in the NPM registry.

### CODE REVIEW

File: client-js/CONTRIBUTING.md [hunk 1]
```diff
@@ -2,17 +2,29 @@
...
 
-The latest published version can be found on [NPM](https://www.npmjs.com/package/spine-web-client).
+The latest published version can be found on [NPM](https://www.npmjs.com/package/spine-web).
 
 ## Usage
 
 To use the library install following modules:
 
 ```
-npm i spine-web-client --save
```
Comment (armiol) [inline client-js/CONTRIBUTING.md]: It's interesting. While reviewing the related `base` changeset I thought we were using `spine-web-client` as an NPM artifact name. However, now I realize we were not. And I like `spine-web` better.

So let's discuss my `base` comment if you need.

File: client-js/main/client/backend-client.js [hunk 2]
```diff
@@ -387,6 +397,29 @@ export class BackendClient {
     return new FirebaseBackendClient(endpoint, firebaseClient, requestFactory, subscriptionService);
   }
 
+
+  /**
+   * Registers all Protobuf types provided by the specified modules.
+   *
+   * After the registration, the types can be used and parsed correctly.
+   *
+   * @param protoIndexFiles the index.js files generated by Spine ProtoJsPlugin
```
Comment (armiol) [inline client-js/main/client/backend-client.js]: Do we call it `ProtoJsPlugin`? Or is it a class? Please review this particular naming here and below.

**Comment by dmytro-grankin** [comment]
@armiol PTAL. The build is failing because `base` was not published.

File: client-js/main/client/parser/well-known-parsers.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    const messageValue = parser.fromObject(packedValue);
+    const bytes = messageValue.serializeBinary();
+    const anyMsg = new Any();
+    anyMsg.setTypeUrl(typeUrl);
+    anyMsg.setValue(bytes);
+    return anyMsg;
+  }
+}
+
+export const wellKnownParsers = new Map([
```
Comment (armiol) [inline client-js/main/client/parser/well-known-parsers.js]: Please make sure the list of supported types is the same we have in `base`. Maybe placing a comment with a reference would help both here and there.

Comment (dmytro-grankin) [reply client-js/main/client/parser/well-known-parsers.js]: As discussed, the parsers have no relation to `base` since they were removed from there.

**Pull request merged**